### ISSUE TITLE
[Impeller] The ListView background image keep flashing when scroll

### ISSUE THREAD

**Issue by techwn** [flutter/flutter#132375]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

[flutter_app.zip](https://github.com/flutter/flutter/files/12319444/flutter_app.zip)
1. download flutter_app.zip
2. exact it and run it with flutter 3.10.6 on iOS (simulator or real device)
3. scroll up/down the page.

### Expected results

I hope the backgroud image can scroll with the list view, It goes well when turn off the Impeller.

### Actual results

the Listview item scroll well, but the decorated background keep flashing and not scroll with the list view.

### Code sample

<details><summary>Code sample</summary>

```dart
CustomScrollView(
        slivers: [
          SliverToBoxAdapter(
            child: Container(
              height: 200,
              width: MediaQueryData.fromView(WidgetsBinding.instance.platformDispatcher.views.first).size.width,
              color: Colors.green,
            ),
          ),
          SliverToBoxAdapter(
            child: Container(
              decoration: BoxDecoration(
                image: DecorationImage(
                  image: Image.asset('assets/bg.webp').image,
                  alignment: Alignment.topCenter,
                  fit: BoxFit.fitWidth,
                  repeat: ImageRepeat.repeatY,
                ),
              ),
              child: ListView.builder(
                physics: const NeverScrollableScrollPhysics(),
                shrinkWrap: true,
                itemBuilder: (context, index) {
                  return ListTile(
                    title: Text("hello $index"),
                  );
                },
                itemCount: 100,
                itemExtent: 100,
              ),
            ),
          )
        ],
      )
```

The core code is DecorationImage, when turn off Impeller on iOS, it works well, but when turn on Impeller, background image keep flashing when scroll.

please use this demo code to run directly:

[flutter_app.zip](https://github.com/flutter/flutter/files/12319444/flutter_app.zip)

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

Flashing video:

https://github.com/flutter/flutter/assets/20474091/51fc9297-b502-4f56-98c0-c539c1be1000

When turn off Impeller, work well video:

https://github.com/flutter/flutter/assets/20474091/900e4028-838c-4d69-8e11-5901cef2f527

</details>

### Logs

No logs for this, please just run the demo code.

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
Last login: Fri Aug 11 15:49:07 on ttys014
qxy9763@LSCN954535 ~ % flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.6, on macOS 13.5 22G74 darwin-x64, locale en-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] IntelliJ IDEA Community Edition (version 2023.2)
[✓] VS Code (version 1.81.0)
[✓] Connected device (4 available)            
[✓] Network resources

• No issues found!
qxy9763@LSCN954535 ~ % 

```

</details>

**Comment by huycozy** [comment]
Hi @techwn 
This issue seems to have been fixed on the latest master channel as I cannot reproduce it there on my device: iPhone 7, iOS 15.7.2.

<details>
<summary>Demo</summary>


https://github.com/flutter/flutter/assets/104349824/bdda28b5-f5dd-4d00-bdfb-e4e7d0a00d42


</details>

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.10.6, on macOS 13.5 22G74 darwin-x64, locale en-VN)
    • Flutter version 3.10.6 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f468f3366c (3 weeks ago), 2023-07-12 15:19:05 -0700
    • Engine revision cdbeda788a
    • Dart version 3.0.6
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 32.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode-beta.app/Contents/Developer
    • Build 15A5195m
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.80.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.68.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 13.5 22G74 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 115.0.5790.114

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

```bash
[!] Flutter (Channel master, 3.14.0-3.0.pre.32, on macOS 13.5 22G74 darwin-x64, locale en-VN)
    • Flutter version 3.14.0-3.0.pre.32 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 653187668a (60 minutes ago), 2023-08-10 22:12:34 -0400
    • Engine revision 578a8e8aab
    • Dart version 3.2.0 (build 3.2.0-58.0.dev)
    • DevTools version 2.26.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 32.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.80.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.70.0

[✓] Connected device (4 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D                         • android-arm64  • Android 11 (API 30)
    • iPhone (mobile)  • d9a94afe2b649fef56ba0bfeb052f0f2a7dae95e • ios            • iOS 15.7.2 19H218
    • macOS (desktop)  • macos                                    • darwin-x64     • macOS 13.5 22G74 darwin-x64
    • Chrome (web)     • chrome                                   • web-javascript • Google Chrome 115.0.5790.170

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>

I'll close this issue as fixed on the latest master channels. If you still experience this issue, kindly leave a comment below and I will reopen this. Thank you!

**Comment by techwn** [comment]
@huycozy,  I also try the master channel, it works fine.
but with stable channel it works not fine.

So will this fix cherry-pick to stable channel as patch fix like: 3.10.7?

We are upgrading our app to latest stable patch fix: 3.10.6, if fix of Impeller not cherry-pick to stable channel, we will need to wait for another 6 months to take the advantage of Impeller in our production.

**Comment by huycozy** [comment]
Please see [Will a particular bug fix be provided in a hotfix release?](https://github.com/flutter/flutter/wiki/Flutter-build-release-channels#will-a-particular-bug-fix-be-provided-in-a-hotfix-release)
